include/swift/SILOptimizer/Utils/DebugOptUtils.h

```diff
@@ -76,6 +81,7 @@ eraseFromParentWithDebugInsts(SILInstruction *inst,
   // Just matching what eraseFromParentWithDebugInsts is today.
   if (nextII == inst->getIterator())
     ++nextII;
+  swift::salvageDebugInfo(inst);
```

Comment:
Does this need to be part of the SROA PR or should this be in its own patch?

Reply:
My original thought was that if salvageDebugInfo is put in a separate patch (or PR), there will be no use case or test there, which is a little weird. What do you think?

Reply:
Could we write a standalone test using sil-opt that runs exactly one pass that would delete an instruction? If the answer is no, then let's just keep it.

Reply:
Currently we're calling salvageDebugInfo inside InstructionDeleter, which will affect multiple Passes. We _can_ first create a PR that only calls salvageDebugInfo in a single Pass then revise it to use InstructionDeleter in later PR...but I feel like it will be a little over-engineering